### DISCUSSION TITLE
Turn off automatic deployments but allow manual triggers

### DISCUSSION THREAD

**Discussion by cj** [vercel/community#257] [Help] [answered]
Hi,

Is it possible to turn off automatic deployments but still allow manually triggering it via the API https://vercel.com/docs/rest-api#endpoints/deployments/create-a-new-deployment?

Cheers!

**Comment by leerob** [comment, marked as answer]
Yes! You can disable the git integration and use the API programmatically if you prefer. Or a deploy hook: https://vercel.com/docs/concepts/git/deploy-hooks

**Reply by cj** [reply]
@leerob if I turn off git integration I won't be able to do:

```typescript
  const response = await client.post('v13/deployments', {
    name: 'system',
    gitSource: {
      type: 'github',
      repoId,
      ref: branchName,
    },
  })
```

There is also no way to force a build and ignore the build ignore command.

**Reply by cj** [reply]
How our workflow works is each pull request creates a full environment, the database and hasura need to be deployed before vercel starts its deployment.  The only issue we are facing is disabling the automatic builds, triggering it manually works great.

**Reply by Pieparker** [reply]
I believe the options are:
- ensure that the PR for your Vercel project is created _after_ your prerequisites are deployed, or
- trigger a deploy of your Vercel project after your prerequisites are deployed, disabling the build which is triggered when the workflow triggers all builds

The question for point 2 is "what can we do to distinguish between the workflow-triggered build and the deploy hook triggered build" in the [ignored build step](https://vercel.com/support/articles/how-do-i-use-the-ignored-build-step-field-on-vercel)? You may want to [check this out](https://github.com/vercel/community/discussions/285#discussioncomment-1696833)!
